tough-kms/src/lib.rs

```diff
@@ -0,0 +1,139 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+mod client;
+pub mod error;
+use rusoto_kms::{Kms, KmsClient};
+use snafu::{OptionExt, ResultExt};
+use std::fmt;
+use tough::key_source::KeySource;
+use tough::schema::decoded::{Decoded, RsaPem};
+use tough::schema::key::KmsSigningAlgorithms;
+use tough::sign::{KmsRsaKey, Sign};
+
+/// Implements the KeySource trait for keys that live in AWS KMS
+pub struct KmsKeySource {
+    pub profile: Option<String>,
+    pub key_id: String,
+    pub client: Option<KmsClient>,
+}
+
+impl fmt::Debug for KmsKeySource {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("KmsKeySource")
+            .field("key_id", &self.key_id)
+            .field("profile", &self.profile)
+            .finish()
+    }
+}
+
+/// Implements the KeySource trait.
+impl KeySource for KmsKeySource {
+    fn as_sign(
+        &self,
+    ) -> std::result::Result<Box<dyn Sign>, Box<dyn std::error::Error + Send + Sync + 'static>>
+    {
+        let kms_client = match self.client.clone() {
```

Comment:
Why do we have to `.clone()` here? I think we should just be able to match on `self.client`, no?

Reply:
Maybe this, unless ownership is actually required.

```suggestion
        let kms_client = &match self.client {
```